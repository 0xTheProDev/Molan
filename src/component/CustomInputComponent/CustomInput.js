// @flow
import React from 'react';
import { Grid, Form, TextArea } from 'semantic-ui-react';
import { noop } from 'util/common';

import { CustomInputInternalProps } from './CustomInput.flow';

export function CustomInput({ theme, CustomInputPlaceholder, onCustomInputChange }: CustomInputInternalProps) {
  return (
    <Grid>
      <Grid.Row columns={1}>
        <Grid.Column stretched>
          <Form>
            <TextArea
              autoHeight
              placeholder={CustomInputPlaceholder}
              onChange={onCustomInputChange}
              className={theme}
            />
          </Form>
        </Grid.Column>
      </Grid.Row>
    </Grid>
  );
}

CustomInput.defaultProps = {
  CustomInputPlaceholder: 'Enter your input here',
  onCustomInputChange: noop,
};
