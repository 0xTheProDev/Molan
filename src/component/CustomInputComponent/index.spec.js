import React from 'react';

import CustomInput from './index';

describe('Component::CustomInput::Integration Suite', () => {
  let
    onCustomInputChangeStub,
    minProps;

  beforeAll(() => {
    onCustomInputChangeStub = jest.fn();

    minProps = {
      theme: 'light',
      onCustomInputChange: onCustomInputChangeStub,
    };
  });

  it('should render with base structure', () => {
    const wrapper = mount(<CustomInput {...minProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke onCustomInputChange callback with new value when text changes', () => {
    const INPUT = 'Test Input';

    const wrapper = mount(<CustomInput {...minProps} />);
    wrapper
      .find('textarea')
      .simulate('change', { target: { value: INPUT }});

    expect(onCustomInputChangeStub).toHaveBeenCalledWith(INPUT);
  });
});
