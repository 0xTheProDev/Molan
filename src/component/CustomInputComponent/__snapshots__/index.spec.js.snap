// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component::CustomInput::Integration Suite should render with base structure 1`] = `
<CustomInputFactory
  onCustomInputChange={[MockFunction]}
  theme="light"
>
  <CustomInput
    CustomInputPlaceholder="Enter your input here"
    onCustomInputChange={[Function]}
    theme="light"
  >
    <Grid>
      <div
        className="ui grid"
      >
        <GridRow
          columns={1}
        >
          <div
            className="one column row"
          >
            <GridColumn
              stretched={true}
            >
              <div
                className="stretched column"
              >
                <Form
                  as="form"
                >
                  <form
                    className="ui form"
                    onSubmit={[Function]}
                  >
                    <TextArea
                      as="textarea"
                      autoHeight={true}
                      className="light"
                      onChange={[Function]}
                      placeholder="Enter your input here"
                      rows={3}
                    >
                      <Ref
                        innerRef={
                          Object {
                            "current": <textarea
                              class="light"
                              placeholder="Enter your input here"
                              rows="3"
                            />,
                          }
                        }
                      >
                        <RefFindNode
                          innerRef={
                            Object {
                              "current": <textarea
                                class="light"
                                placeholder="Enter your input here"
                                rows="3"
                              />,
                            }
                          }
                        >
                          <textarea
                            autoHeight={true}
                            className="light"
                            onChange={[Function]}
                            onInput={[Function]}
                            placeholder="Enter your input here"
                            rows={3}
                          />
                        </RefFindNode>
                      </Ref>
                    </TextArea>
                  </form>
                </Form>
              </div>
            </GridColumn>
          </div>
        </GridRow>
      </div>
    </Grid>
  </CustomInput>
</CustomInputFactory>
`;
